### PR TITLE
configure settings.xml during build to define USB not supported

### PR DESCRIPTION
**Pull request by AlwinEsch**
This change it to settings.xml.in and set on add-on compile the settings usable or not. Is done by change the `<visible>` value and set of default connection type. The settings are still inside to prevent problems if something tries somewhere to read it.

Further can for build the cmake configuration value "-DUSB_DEVICE_SUPPORT=0" used and if wanted to disable USB support.

XML with USB support disabled
```xml
        <!-- USB device not suppoted on this add-on, due to this the following setting is not usable and marked as not visible. -->
        <setting id="device_connection" type="integer" label="30100" help="30500">
          <level>0</level>
          <visible>false</visible>
          <default>1</default>
          <constraints>
            <options>
              <option label="30200">0</option>
              <option label="30201">1</option>
            </options>
          </constraints>
          <control type="spinner" format="integer"/>
        </setting>

        <!-- USB device not suppoted on this add-on, due to this the following setting is not usable and marked as not visible. -->
        <setting id="device_connection_usb_index" parent="device_connection" type="integer" label="30101" help="30501">
          <level>0</level>
          <visible>false</visible>
          <default>0</default>
          <constraints>
            <minimum>0</minimum>
            <step>1</step>
            <maximum>8</maximum>
          </constraints>
          <dependencies>
            <dependency type="visible" setting="device_connection" operator="is">0</dependency>
          </dependencies>
          <control type="spinner" format="integer"/>
        </setting>

        <setting id="device_connection_tcp_host" parent="device_connection" type="string" label="30102" help="30502">
          <level>0</level>
          <default/>
          <constraints>
            <allowempty>true</allowempty>
          </constraints>
          <!--
          <dependencies>
            <dependency type="visible" setting="device_connection" operator="is">1</dependency>
          </dependencies>
          -->
          <control type="edit" format="ip">
            <heading>30102</heading>
          </control>
        </setting>

        <setting id="device_connection_tcp_port" parent="device_connection" type="integer" label="30103" help="30503">
          <level>0</level>
          <default>1234</default>
          <constraints>
            <allowempty>false</allowempty>
            <minimum>0</minimum>
            <maximum>65535</maximum>
          </constraints>
          <!--
          <dependencies>
            <dependency type="visible" setting="device_connection" operator="is">1</dependency>
          </dependencies>
          -->
          <control type="edit" format="integer">
            <heading>30103</heading>
          </control>
        </setting>
```

XML with USB support enabled
```xml
        <!-- USB device supported on this add-on. -->
        <setting id="device_connection" type="integer" label="30100" help="30500">
          <level>0</level>
          <visible>true</visible>
          <default>0</default>
          <constraints>
            <options>
              <option label="30200">0</option>
              <option label="30201">1</option>
            </options>
          </constraints>
          <control type="spinner" format="integer"/>
        </setting>

        <!-- USB device supported on this add-on. -->
        <setting id="device_connection_usb_index" parent="device_connection" type="integer" label="30101" help="30501">
          <level>0</level>
          <visible>true</visible>
          <default>0</default>
          <constraints>
            <minimum>0</minimum>
            <step>1</step>
            <maximum>8</maximum>
          </constraints>
          <dependencies>
            <dependency type="visible" setting="device_connection" operator="is">0</dependency>
          </dependencies>
          <control type="spinner" format="integer"/>
        </setting>
```

### CODE REVIEW

File: pvr.rtlradio/resources/settings.xml.in [hunk 1]
```diff
@@ -53,9 +59,11 @@
             <minimum>0</minimum>
             <maximum>65535</maximum>
           </constraints>
+          @SETTINGS_XML_IN_NO_USB_BEGIN@
           <dependencies>
             <dependency type="visible" setting="device_connection" operator="is">1</dependency>
           </dependencies>
+          @SETTINGS_XML_IN_NO_USB_END@
```
Comment (AlwinEsch) [inline pvr.rtlradio/resources/settings.xml.in:70]: I just added this as an afterthought. Found the case that USB was previously activated during the test and this was no longer displayed.

File: src/addon.cpp [hunk 2]
```diff
@@ -1153,8 +1153,12 @@ ADDON_STATUS addon::Create(void)
       }
 
       // Load the device settings
+#ifdef USB_DEVICE_SUPPORT
       m_settings.device_connection =
           kodi::addon::GetSettingEnum("device_connection", device_connection::usb);
+#else
+      m_settings.device_connection = device_connection::rtltcp;
+#endif
```
Comment (AlwinEsch) [inline src/addon.cpp:1164]: Because USB was enabled, it is now forced to use tcp.

**Comment by AlwinEsch** [comment]
Maybe another settings text could be good🤔?
Using “- rtl_sdr” as the start looks kind of weird.
Screenshot 2024-04-03 164704

**Comment by AlwinEsch** [comment]
Have added second commit to add groups and to remove the "- ", further renamed with remove of "rtl_tcp " in text.
Screenshot 2024-04-03 171850
Screenshot 2024-04-03 172054
Screenshot 2024-04-03 172239

File: CMakeLists.txt [hunk 3]
```diff
@@ -4,11 +4,25 @@ project(pvr.rtlradio)
...
 if(CORE_SYSTEM_NAME STREQUAL windowsstore)
   set(USB_DEVICE_SUPPORT False)
 else()
-  set(USB_DEVICE_SUPPORT True)
+  option(USB_DEVICE_SUPPORT "Build add-on with USB device support, if set to \"OFF\" RTL-SDR can only be accessed via network" ON)
+endif()
+
+# Set values used about configure of settings.xml.in on add-on
+if(NOT USB_DEVICE_SUPPORT)
+  set(SETTINGS_XML_IN_CONNECTION_TYPE_NOTE "USB device not suppoted on this add-on, due to this the following setting is not usable and marked as not visible.")
```
Comment (ksooo) [inline CMakeLists.txt]: Typo: suppoted -> supported

File: src/addon.cpp [hunk 4]
```diff
@@ -1153,8 +1153,12 @@ ADDON_STATUS addon::Create(void)
...
       m_settings.device_connection_usb_index =
           kodi::addon::GetSettingInt("device_connection_usb_index", 0);
```
Comment (ksooo) [inline src/addon.cpp]: You only need to read this property value if `USB_DEVICE_SUPPORT`is defined.

Comment (AlwinEsch) [reply src/addon.cpp]: Have changed there, also on the SettingsChange function is now the `USB_DEVICE_SUPPORT`

File: CMakeLists.txt (hunk 3 above)
Comment (AlwinEsch) [reply CMakeLists.txt]: Thanks, always somewhere a typo 😅.

**Review by ksooo — CHANGES_REQUESTED** [review]
addon.cpp does not compile if `USB_DEVICE_SUPPORT` is not defined. Look at line 1440, there is an 'else if' without an 'if' then.

**Comment by ksooo** [comment]
Shouldn't `Jenkinsfile` be adjusted to build all platforms again?

**Comment by AlwinEsch** [comment]
> Shouldn't `Jenkinsfile` be adjusted to build all platforms again?

Yes, now all OS works, have added a commit about.

**Comment by AlwinEsch** [comment]
> addon.cpp does not compile if `USB_DEVICE_SUPPORT` is not defined. Look at line 1440, there is an 'else if' without an 'if' then.

Ups, is fixed now. But Jenkins brings now the next and iOS/tvOS become USB device support removed too.
It brings now:
```
In file included from os/darwin_usb.c:50:

os/darwin_usb.h:28:10: fatal error: 'IOKit/IOTypes.h' file not found

#include <IOKit/IOTypes.h>
```
Seems something updated in Jenkins on the morning.

**Comment by AlwinEsch** [comment]
On new added commit becomes the USB support on Android removed too and Jenkins now happy again on all OS.
EDIT: Happy on all OS after several additional commits.

**Review by ksooo — APPROVED** [review]
LGTM

**Pull request merged**